vllm/transformers_utils/s3_utils.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import fnmatch
+import os
+import shutil
+import signal
+import tempfile
+from pathlib import Path
+
+import boto3
+
+
+def _filter_allow(paths: list[str], patterns: list[str]) -> list[str]:
+    return [
+        path for path in paths if any(
+            fnmatch.fnmatch(path, pattern) for pattern in patterns)
+    ]
+
+
+def _filter_ignore(paths: list[str], patterns: list[str]) -> list[str]:
+    return [
+        path for path in paths
+        if not any(fnmatch.fnmatch(path, pattern) for pattern in patterns)
+    ]
+
+
+def glob(s3=None,
+         path: str = "",
+         allow_pattern: list[str] | None = None) -> list[str]:
```

Comment:
"|" is supported since 3.10 and vLLM still supports 3.9, so please use Optional for now.